### PR TITLE
add move_anchor dispatch

### LINKED ISSUE #84: Schedule for Commit
```json=
{
    "anchorID": currentVersion
}
```
- Validates that anchor exists in centchain
- Emits event to be processed by bridge operators with anchorID-docRoot

### PR DESCRIPTION
**Pull request by vedhavyas**
Once this is in, I will add the functional tests.

Closes #84

### CODE REVIEW

File: runtime/src/anchor/mod.rs [hunk 1]
```diff
@@ -91,6 +105,8 @@ decl_storage! {
 decl_module! {
     pub struct Module<T: Trait> for enum Call where origin: T::Origin {
 
+        fn deposit_event<T>() = default;
```
Comment (mikiquantum) [inline runtime/src/anchor/mod.rs]: What is the purpose of this? Apologize in advance for my lack of knowledge on RUST/Substrate

File: runtime/src/anchor/mod.rs [hunk 2]
```diff
@@ -237,6 +253,17 @@ decl_module! {
 
             Ok(())
         }
+
+        // move_anchor dispatch call when anchor by anchor_id is to be moved to another chain
+        pub fn move_anchor(origin, anchor_id: T::Hash) -> Result {
+            // ensure signed origin
+            ensure_signed(origin)?;
```
Comment (mikiquantum) [inline runtime/src/anchor/mod.rs]: Is this origin the validator signature? or the client/requester signature?

**Comment by vedhavyas** [comment]
> Since we run the bridges and they should take care of not doing actions twice (for $$ reasons). And we do not keep track of the moved anchors yet, the bridges should take care of what anchors have been processed already (for a given anchorID), so we do not end up in a DDOS attack and bridge funds are wiped.
Just to keep in mind, for the bridge design.

I have discussed these with @vimukthi-git as well. For the first version, I think we should be good. We anyway discard this as soon as we start implemeting bucket logic

File: runtime/src/anchor/mod.rs (hunk 1 above)
Comment (vedhavyas) [reply runtime/src/anchor/mod.rs]: this is the module function that is used to send an event. `default` implies we are choosing to use the substrate's default event triggering logic

File: runtime/src/anchor/mod.rs (hunk 2 above)
Comment (vedhavyas) [reply runtime/src/anchor/mod.rs]: This validates the sender of the dispatch call. Ethereum does it out of the box but in substrate different accounts can send a dispatch call. For this function `ensure-signed` is enough. For sudo users, we would be using a different fucntion. 

@vimukthi-git can shed more light on this maybe ?

Comment (vimukthi-git) [reply runtime/src/anchor/mod.rs]: This is basically the signer of the extrinsic, or the entity who originated the call. Ensure signed is used whenever we want to make sure that the extrinsic is signed. Signing account is the account which pays the transaction fees if any also.

Comment (mikiquantum) [reply runtime/src/anchor/mod.rs]: That means that the chain needs to know the public key to verify the signature. Is that a previous setup needed for each account? Or it is implicit like in Ethereum where public key is the address (simplifying)? Or the public key goes along the `origin` object?

Comment (vedhavyas) [reply runtime/src/anchor/mod.rs]: Its like ethereum i guess.

Comment (vimukthi-git) [reply runtime/src/anchor/mod.rs]: Yup its like ethereum, account addresses are public keys.

**Pull request merged**